src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,123 @@
+    public ArrayList() {
+        this.elementData = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size >= elementData.length) {
+            elementData = growIfArrayFull();
+        }
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("index doesn't exist");
+        }
+        if (size >= elementData.length) {
+            elementData = growIfArrayFull();
+        }
+        //shift of positions to the right side
+        for (int i = size; i > index; i--) {
+            elementData[i] = elementData[i - 1];
+        }
+        elementData[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        findIndex(index);
+        return elementData[index];
+
     }
 
     @Override
     public void set(T value, int index) {
+        findIndex(index);
+        elementData[index] = value;
+
+    }
 
+    private void findIndex(int index) {
+        if (index < 0 || index > size || index >= size) {
+            throw new ArrayListIndexOutOfBoundsException("index doesn't exist");
+        }
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        findIndex(index);
+        final T value = elementData[index];
+        //shift of positions to the left side
+        for (int i = index; i < size - 1; i++) {
+            elementData[i] = elementData[i + 1];
+        }
+        elementData[size - 1] = null;
+        size--;
+        return value;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        int indexOfElement = -1;
+
+        for (int i = 0; i < size; i++) {
+            if (elementData[i] != null && elementData[i].equals(element)) {
+                indexOfElement = i;
+                break;
+            } else if (elementData[i] == null) {
+                indexOfElement = i;
+                break;
+            }
+        }
+        if (indexOfElement == -1) {
+            throw new NoSuchElementException("No such element");
+        }
+        //shift of positions to the left side
+        for (int i = indexOfElement; i < size - 1; i++) {
+            elementData[i] = elementData[i + 1];
+        }
+        elementData[size - 1] = null;
+        size--;
+        return element;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    @SuppressWarnings("unchecked")
+    private T[] growIfArrayFull() {
+        int newSize = (int) (elementData.length * 1.5);
```

Comment:
Magic number '1.5' should be replaced with a named constant to make the code more readable and maintainable.